File: C++/Medium/1653.cc
#include <string>
#include <algorithm>

using namespace std;

class Solution {
public:
    int minimumDeletions(string s) {
        int a_right = count(s.begin(), s.end(), 'a');
        int b_left = 0;
        int ans = a_right;

        for (int i = 0; i < s.size(); i++){
            if (s[i] == 'a'){
                a_right -= 1;
            }else if (s[i] == 'b'){
                b_left += 1;
            }
            ans = min(ans, a_right + b_left);
        }
        return ans;
    }
};

File: Python/Medium/1653.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        a_right = s.count('a')
        b_left = 0
        ans = a_right
        for ch in s:
            if ch == 'a':
                a_right -= 1
            elif ch == 'b':
                b_left += 1
            ans = min(ans, a_right + b_left)
        return ans 
        
